src/hooks.ts

```diff
@@ -36,3 +45,110 @@ export const useHasResourcePermission = (resource: Resource, authzUrl: string, p
     const { permissions, isFetching } = useResourcePermissions(resource, authzUrl) ?? {};
     return { isFetching, hasPermission: permissions.includes(permission) };
 };
+
+export const useOpenIdConfig = (openIdConfigUrl: string) => {
+    const dispatch = useDispatch();
+
+    dispatch(fetchOpenIdConfigurationIfNecessary(openIdConfigUrl));
```

Comment:
this should still be in a `useEffect` depending on `dispatch` and `openIdConfigUrl` I think, since otherwise the if necessary function is called way more than needed which means the action dispatch log will be pretty hard to debug if we need to